Assets/Scripts/Models/Buildable/Components/PowerConnection.cs

```diff
@@ -0,0 +1,259 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Xml.Serialization;
+using MoonSharp.Interpreter;
+using ProjectPorcupine.PowerNetwork;
+
+namespace ProjectPorcupine.Buildable.Components
+{
+    [Serializable]
+    [XmlRoot("Component")]
+    [BuildableComponentName("PowerConnection")]
+    [MoonSharpUserData]
+    public class PowerConnection : BuildableComponent, IPlugable
+    {       
+        public PowerConnection()
+        {
+        }
+
+        private PowerConnection(PowerConnection other) : base(other)
+        {
+            ParamsDefinitions = other.ParamsDefinitions;
+            Provides = other.Provides;
+            Requires = other.Requires;
+        }
+
+        [XmlElement("ParameterDefinitions")]
+        public PowerConnectionParameterDefinitions ParamsDefinitions { get; set; }
+
+        public Parameter CurrentAccumulatorCharge
+        {
+            get
+            {
+                return FurnitureParams[ParamsDefinitions.CurrentAcumulatorCharge.ParameterName];
+            }
+        }
+
+        public Parameter CurrentAccumulatorChargeIndex
+        {
+            get
+            {
+                return FurnitureParams[ParamsDefinitions.CurrentAcumulatorChargeIndex.ParameterName];
+            }
+        }
+
+        [XmlIgnore]
+        public bool IsRunning
+        {
+            get
+            {
+                return FurnitureParams[ParamsDefinitions.IsRunning.ParameterName].ToBool();
+            }
+
+            set
+            {
+                FurnitureParams[ParamsDefinitions.IsRunning.ParameterName].SetValue(value);
+            }
+        }
+
+        [XmlElement("Provides")]
+        public Info Provides { get; set; }
+
+        [XmlElement("Requires")]
+        public Info Requires { get; set; }
+        
+        [XmlIgnore]
+        public float AccumulatedAmount
+        {
+            get
+            {
+                return CurrentAccumulatorCharge.ToFloat();
+            }
+
+            set
+            {
+                float oldAccPower = CurrentAccumulatorCharge.ToFloat();
+                if (oldAccPower != value)
```

Comment:
I'm on the fence of rather or not this variable name needs changed, it may be enough that the purpose of the variable is clear from seeing just the name, but I'm not absolutely sure.